library/src/main/java/com/wultra/android/mtokensdk/oidc/utils/OIDCUtils.kt

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2025 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.wultra.android.mtokensdk.oidc.utils
+
+import android.net.Uri
+import android.util.Base64
+import com.wultra.android.mtokensdk.log.WMTLogger
+import com.wultra.android.mtokensdk.oidc.models.OIDCAuthorizationRequest
+import com.wultra.android.mtokensdk.oidc.models.OIDCConfig
+import com.wultra.android.mtokensdk.oidc.models.OIDCPowerAuthActivationAttributes
+import com.wultra.android.mtokensdk.oidc.models.PKCECodes
+import java.security.MessageDigest
+import java.security.SecureRandom
+
+/** Utility object for OIDC-related operations, such as PKCE generation and URI handling. */
+object OIDCUtils {
```

Comment:
We use `class` usually for Utils classes (for example `PACUtils`) with companion object. We should be consistent.